keps/sig-auth/3221-structured-authorization-configuration/README.md

```diff
@@ -0,0 +1,802 @@
+
+- [x] Feature gate
+  - Feature gate name: `AuthorizationConfigFromFile`
+  - Components depending on the feature gate:
+        - kube-apiserver
+
+###### Does enabling the feature change any default behavior?
+
+No.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled once enabled.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+When the feature is re-enabled, `--authorization-config-file` flag should be present.
+The behaviour is the same as when the feature is enabled for the first time.
+
+###### Are there any tests for feature enablement/disablement?
+
+We will have extensive unit tests during feature implementation. There would be tests
+for the Authorizer chain in both the old and new configuration scenarios.
+
+We will add integration tests to validate the enablement/disablement flow.
+  - When the feature is disabled, only the existing command line flag `--authorization-webhook-*`
+based mode is allowed.
+  - When the feature is enable, setting both `--authorization-config-file` and
+configuring an authorization webhook using the `--authorization-webhook-*`
+command line flags should return an error.
+
+### Rollout, Upgrade and Rollback Planning
+
+> Note: This section is required when targeting Beta to a release.
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A rollout can fail when the authorization configuration file being passed doesn't
+exist or is invalid.
+
+Already running workloads won't be impacted but cluster users won't be able to
+access the control plane if the cluster is single-node.
+
+###### What specific metrics should inform a rollback?
+
+Not applicable.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+TBD.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+None.
+
+### Monitoring Requirements
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+> Note: To be elaborated more during Beta graduation since this section
+must be completed when targeting beta to a release.
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The cluster administrators can check the flags passed to the `kube-apiserver` if
+they have access to the control plane nodes. If the `--authorization-config-file`
+is set to a valid authorization configuration file, the feature is being used.
+Or, they can look at the metrics exposed by `kube-apiserver`.
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Other
+  - Details: They can look at the metrics if `apiserver_authorization_decisions_total` is increasing.
```

Comment:
This would increment even if this feature was not in use (this is a pre-existing metric right?).

Reply:
This is a new metric similar to all other metrics mentioned in this KEP.

Reply:
I see the confusion now. I have updated to:
> Details: Since this feature introduced the `name` field to the webhook authorizer,
  users can first specify a value in the `name` field of the AuthorizationConfiguration.
  Then look at the `apiserver_authorization_webhook_evaluations_total` metrics to ensure the
  count for the named webhook authorizer is increasing.